src/libcommonserver/log/log.cpp

```diff
@@ -109,4 +115,393 @@ Log::Log(const log4cplus::tstring &filePath) {
+                 "Error in IoHelper::copyFileOrDirectory : "
+                     << Utility::formatIoError(logPath / "send_log_directory_temp" / archiveName, ioError).c_str());
+        IoHelper::deleteDirectory(tempDirectory.parent_path(), ioError);
+
+        exitCause = ExitCauseUnknown;
+        return ExitCodeUnknown;
+    }
+
+    // Delete the temp folder
+    if (!IoHelper::deleteDirectory(tempDirectory.parent_path(), ioError)) {
+        LOG_WARN(Log::instance()->getLogger(),
+                 "Error in IoHelper::deleteDirectory: " << Utility::formatIoError(tempDirectory.parent_path(), ioError).c_str());
+    }
+
+    exitCause = ExitCauseUnknown;
+    return ExitCodeOk;
+}
+
+ExitCode Log::copyParmsDbTo(const SyncPath &outputPath, ExitCause &exitCause) {
+    const SyncPath parmsDbName = ".parms.db";
+    const SyncPath parmsDbPath = CommonUtility::getAppSupportDir() / parmsDbName;
+    DirectoryEntry entryParmsDb;
+    IoError ioError = IoErrorUnknown;
+    exitCause = ExitCauseUnknown;
+
+    if (!IoHelper::getDirectoryEntry(parmsDbPath, ioError, entryParmsDb)) {
+        LOG_WARN(Log::instance()->getLogger(),
+                 "Error in IoHelper::getDirectoryEntry: " << Utility::formatIoError(parmsDbPath, ioError).c_str());
+        if (ioError == IoErrorNoSuchFileOrDirectory) {
+            exitCause = ExitCauseFileAccessError;
+        } else {
+            exitCause = ExitCauseUnknown;
+        }
+        return ExitCodeSystemError;
+    }
+
+    if (!IoHelper::copyFileOrDirectory(parmsDbPath, outputPath, ioError)) {
+        LOG_WARN(Log::instance()->getLogger(),
+                 "Error in IoHelper::copyFileOrDirectory: " << Utility::formatIoError(parmsDbPath, ioError).c_str());
+
+        if (ioError == IoErrorDiskFull) {
+            exitCause = ExitCauseNotEnoughDiskSpace;
+        } else {
+            exitCause = ExitCauseUnknown;
+        }
+        return ExitCodeSystemError;
+    }
+    exitCause = ExitCauseUnknown;
+    return ExitCodeOk;
+}
+
+ExitCode Log::copyLogsTo(const SyncPath &outputPath, bool includeOldLogs, ExitCause &exitCause) {
+    SyncPath logPath = _filePath.parent_path();
+    IoError ioError = IoErrorUnknown;
+    IoHelper::DirectoryIterator dir;
+
+    if (!IoHelper::getDirectoryIterator(logPath, false, ioError, dir)) {
+        LOG_WARN(Log::instance()->getLogger(),
+                 "Error in DirectoryIterator: " << Utility::formatIoError(logPath, ioError).c_str());
+        exitCause = ExitCauseUnknown;
+        return ExitCodeSystemError;
+    }
+
+    DirectoryEntry entry;
+    bool endOfDirectory = false;
+    while (dir.next(entry, endOfDirectory, ioError) && !endOfDirectory) {
+        if (entry.is_directory()) {
+            LOG_WARN(Log::instance()->getLogger(), "Ignoring temp directory " << entry.path().filename().string().c_str());
+            continue;
+        }
+
+        if (!includeOldLogs && entry.path().filename().extension() == L".gz") {
```

Comment:
`filename()`return a `wstring`on Windows but not in Unix. Perhaps convert the string ".gz".
I think there is a method in Utilits named `Str()`that does it. for you